tests/test-basic-user-only.sh

```diff
@@ -54,6 +54,16 @@ fi
 assert_file_has_content err.txt "Content object.*invalid mode.*with bits 040.*"
 echo "ok failed to commit suid"
 
+cd ${test_tmpdir}
+rm repo-input -rf
+ostree_repo_init repo-input init --mode=archive
+rm files -rf && mkdir files
+if $CMD_PREFIX ostree --repo=repo-input commit -b willfail --tree=dir=files --add-metadata-string='=FOO'; then
+    assert_not_reached "committed an empty metadata key"
+fi
```

Comment:
Minor/optional: I like to also verify the error message is what we expect even if it makes the test slightly more implementation-dependent. This way you're sure that you're testing the checks you think you are and won't unknowingly regress if it turns out you weren't.